vignettes/ProgenySingleCell.Rmd

```diff
@@ -0,0 +1,243 @@
+`r CRANpkg("Seurat")` vignette before going deeper into the data analysis. 
+These steps carry out the selection and filtration of cells based on quality
+control metrics, the data normalization and scaling, and the detection of highly
+variable features (see https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html). 
+
+```{r, message=FALSE}
+## Identification of mithocondrial genes
+pbmc[["percent.mt"]] <- PercentageFeatureSet(pbmc, pattern = "^MT-")
+
+## Filtering cells following standard QC criteria.
+pbmc <- subset(pbmc, subset = nFeature_RNA > 200 & nFeature_RNA < 2500 & 
+    percent.mt < 5)
+
+## Normalizing the data
+pbmc <- NormalizeData(pbmc, normalization.method = "LogNormalize", 
+    scale.factor = 10000)
+
+pbmc <- NormalizeData(pbmc)
+
+## Identify the 2000 most highly variable genes
+pbmc <- FindVariableFeatures(pbmc, selection.method = "vst", nfeatures = 2000)
+
+## In addition we scale the data
+all.genes <- rownames(pbmc)
+pbmc <- ScaleData(pbmc, features = all.genes)
+```
+
+## Clustering cells
+
+One of the most relevant steps in scRNA-seq data analysis is clustering. Cells
+are grouped based on the similarity of their transcriptomic profiles. We first 
+apply the `r CRANpkg("Seurat")` v3 classical approach as described in their 
+aforementioned vignette. We visualize the cell clusters using UMAP: 
+
+```{r, message=FALSE, warning=FALSE}
+pbmc <- 
+    RunPCA(pbmc, features = VariableFeatures(object = pbmc), verbose = FALSE)
+pbmc <- FindNeighbors(pbmc, dims = 1:10, verbose = FALSE)
+pbmc <- FindClusters(pbmc, resolution = 0.5, verbose = FALSE)
+pbmc <- RunUMAP(pbmc, dims = 1:10,  umap.method = 'umap-learn',  
+    metric='correlation')
+```
+
+```{r}
+DimPlot(pbmc, reduction = "umap")
+```
+
+## Pathway activity per cell population
+
+Following again `r CRANpkg("Seurat")` protocol, we next find the markers 
+that help to identify and discriminate the different cell popualations 
+present in the dataset under study.
+
+```{r, message = FALSE}
+## Finding differentially expressed features (cluster biomarkers)
+pbmc.markers <- FindAllMarkers(pbmc, only.pos = TRUE, min.pct = 0.25, 
+    logfc.threshold = 0.25, verbose = FALSE)
+
+## Assigning cell type identity to clusters
+new.cluster.ids <- c("Naive CD4 T", "Memory CD4 T", "CD14+ Mono", "B", "CD8 T", 
+    "FCGR3A+ Mono", "NK", "DC", "Platelet")
+names(new.cluster.ids) <- levels(pbmc)
+pbmc <- RenameIdents(pbmc, new.cluster.ids)
+
+## We create a data frame with the specification of the cells that belong to 
+## each cluster to match with the Progeny scores.
+CellsClusters <- data.frame(Cell = names(Idents(pbmc)), 
+    CellType = as.character(Idents(pbmc)),
+    stringsAsFactors = FALSE)
+```
+
+We plot again the clusters along with their cell type labels
+
+```{r, message = FALSE, warning = FALSE}
+#DimPlot(pbmc, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
+```
+
+Finally, we compute **PROGENy** pathway activity scores on the scRNA-seq data, 
```

Comment:
Maybe here you should update the vignette to the new feature that progeny works directly with Seurat objects